test/integration/single_user_test.go

```diff
@@ -134,5 +136,34 @@ func TestSingleUser(t *testing.T) {
 		},
 	}
 	expectedEvents := []string{"", "test-pod\n"}
-	testutil.AssertLogsForExec(t, logs, "/api/v1/namespaces/default/pods/test-pod/exec?command=cat&command=%2Fetc%2Fhostname&container=test-pod&stderr=true&stdout=true", expectedUser, expectedHeader, expectedEvents)
+	testutil.AssertLogsForExecOrAttach(t, logs, "/api/v1/namespaces/default/pods/test-pod/exec?command=cat&command=%2Fetc%2Fhostname&container=test-pod&stderr=true&stdout=true", expectedUser, expectedHeader, expectedEvents)
+
+	// Test `kubectl attach`
+	expectedHeader = wsproxy.AsciicastHeader{
+		Version:   2,
+		Width:     0,
+		Height:    0,
+		Timestamp: 0,
+		User:      expectedUser["username"].(string),
+		K8sMetadata: &wsproxy.K8sMetadata{
+			PodName:   "test-pod",
+			Namespace: "default",
+			Container: "test-pod",
+		},
+	}
+	expectedEvents = []string{"", "hello\n"}
+
+	var exitError *exec.ExitError
+
+	output, err = user.Kubectl.CommandWithTimeout(time.Second, "attach", "test-pod")
+	// ExitError is expected here because the command gets killed because of the timeout
+	require.ErrorAs(t, err, &exitError)
+	require.Empty(t, exitError.Stderr, "failed to execute kubectl attach")
+
+	assert.Contains(t, string(output), "hello\n")
+
+	// Wait for the logs to be flushed
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Just FYI, when Go 1.25 lands later this year, we should try to use [`synctest`](https://antonz.org/go-1-25/#synthetic-time-for-testing) to avoid these sleeps.